app/src/main/java/com/github/se/orator/model/chatGPT/ChatViewModel.kt

```diff
@@ -30,60 +32,60 @@ class ChatViewModel(
   private val _errorMessage = MutableStateFlow<String?>(null)
   val errorMessage = _errorMessage.asStateFlow()
 
-  private val _collectedAnalysisData = MutableStateFlow<MutableList<AnalysisData>>(mutableListOf())
+  // Change to List<AnalysisData> and update properly
+  private val _collectedAnalysisData = MutableStateFlow<List<AnalysisData>>(emptyList())
```

Comment:
Hi, it was made to ensure that updates to the data are properly propagated through the StateFlow to any observers because when you use a MutableStateFlow, it only emits new values when its value property is updated (for ex when the reference changes). If you have a mutable object (like a MutableList) and you modify it in place ( if you add or remove items), the reference to the list remains the same. As a result, MutableStateFlow does not detect any changes because the value hasn't been reassigned, it's still pointing to the same list instance.

Reply:
Thank you, it makes sense now !